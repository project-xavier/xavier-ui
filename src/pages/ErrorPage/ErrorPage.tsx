import React from 'react';
import { Link } from 'react-router-dom';
import {
  Bullseye,
  EmptyState,
  EmptyStateVariant,
  EmptyStateIcon,
  Title,
  EmptyStateBody,
} from '@patternfly/react-core';
import { ErrorCircleOIcon } from '@patternfly/react-icons';
import {
  PageHeader,
  PageHeaderTitle,
} from '@redhat-cloud-services/frontend-components/components/PageHeader';
import { Main } from '@redhat-cloud-services/frontend-components/components/Main';

class ErrorPage extends React.Component {
  public render() {
    return (
      <React.Fragment>
        <PageHeader>
          <PageHeaderTitle title={'Error'} />
        </PageHeader>
        <Main>
          <Bullseye>
            <EmptyState variant={EmptyStateVariant.large}>
              <EmptyStateIcon icon={ErrorCircleOIcon} />
              <Title headingLevel="h5" size="lg">
                Error
              </Title>
              <EmptyStateBody>Something unexpected happened, please try again!</EmptyStateBody>
              <Link to={'/'} className="pf-c-button pf-m-primary" target="_self">
                Home
              </Link>
            </EmptyState>
          </Bullseye>
        </Main>
      </React.Fragment>
    );
  }
}

export default ErrorPage;
