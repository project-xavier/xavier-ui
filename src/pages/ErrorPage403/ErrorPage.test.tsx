import React from "react";
import { shallow } from "enzyme";
import ErrorPage from "./ErrorPage403";

describe("ErrorPage403", () => {

  it("expect to render", () => {
    const wrapper = shallow(
      <ErrorPage />
    );
    expect(wrapper).toMatchSnapshot();
  });

});
