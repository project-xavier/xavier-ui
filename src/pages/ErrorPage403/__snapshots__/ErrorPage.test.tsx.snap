// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage403 expect to render 1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      title="Error: 403 Forbidden"
    />
  </PageHeader>
  <Connect(Main)>
    <Bullseye>
      <EmptyState
        variant="large"
      >
        <EmptyStateIcon
          icon={[Function]}
        />
        <Title
          headingLevel="h5"
          size="lg"
        >
          Error: 403 Forbidden
        </Title>
        <EmptyStateBody>
          You tried to access a forbidden resource!.
        </EmptyStateBody>
        <Link
          className="pf-c-button pf-m-primary"
          replace={false}
          target="_self"
          to="/"
        >
          Home
        </Link>
      </EmptyState>
    </Bullseye>
  </Connect(Main)>
</Fragment>
`;
